### PR TITLE
Makes all toy figurines tiny and refactors the code behind it

### PR DESCRIPTION
**Pull request by farie82**
## What Does This PR Do
As the title says. Makes all toy figurines tiny and refactors the code behind it to be better

## Why It's Good For The Game
Some toy figurines being normal sized is a bit odd. Them being tiny isn't a mechanical advantage and will make carrying one around in your pocket possible

## Changelog
:cl:
tweak: All toy figurines are now tiny!
fix: All crew toys now have unique descriptions as intended
/:cl:

### CODE REVIEW

**Comment by Wojz** [comment]
Now I get to use the captain figure as captain whenever I please!

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1570,243 +1413,387 @@
...
-/obj/item/toy/figure/borg
+/obj/item/toy/figure/crew/borg
 	name = "Cyborg action figure"
 	desc = "The iron-willed cyborg, from Space Life's SS12 figurine collection."
 	icon_state = "borg"
 	toysay = "I. LIVE. AGAIN."
 
-/obj/item/toy/figure/botanist
+/obj/item/toy/figure/crew/botanist
 	name = "Botanist action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1484]:
```suggestion
	name = "\improper Botanist action figure"
```

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The drug-addicted botanist, from Space Life's SS12 figurine collection."
 	icon_state = "botanist"
 	toysay = "Dude, I see colors..."
 
-/obj/item/toy/figure/captain
+/obj/item/toy/figure/crew/captain
 	name = "Captain action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1490]:
```suggestion
	name = "\improper Captain action figure"
```

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The inept captain, from Space Life's SS12 figurine collection."
 	icon_state = "captain"
 	toysay = "Crew, the Nuke Disk is safely up my ass."
 
-/obj/item/toy/figure/cargotech
+/obj/item/toy/figure/crew/cargotech
 	name = "Cargo Technician action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1496]:
```suggestion
	name = "\improper Cargo Technician action figure"
```

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The hard-working cargo tech, from Space Life's SS12 figurine collection."
 	icon_state = "cargotech"
 	toysay = "For Cargonia!"
 
-/obj/item/toy/figure/ce
+/obj/item/toy/figure/crew/ce
 	name = "Chief Engineer action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1502]:
```suggestion
	name = "\improper Chief Engineer action figure"
```

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The expert Chief Engineer, from Space Life's SS12 figurine collection."
 	icon_state = "ce"
 	toysay = "Wire the solars!"
 
-/obj/item/toy/figure/chaplain
+/obj/item/toy/figure/crew/chaplain
 	name = "Chaplain action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1508]:
```suggestion
	name = "\improper Chaplain action figure"
```

File: code/game/objects/items/toys.dm [hunk 6]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The obsessed Chaplain, from Space Life's SS12 figurine collection."
 	icon_state = "chaplain"
 	toysay = "Gods make me a killing machine please!"
 
-/obj/item/toy/figure/chef
+/obj/item/toy/figure/crew/chef
 	name = "Chef action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1514]:
```suggestion
	name = "\improper Chef action figure"
```

File: code/game/objects/items/toys.dm [hunk 7]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The cannibalistic chef, from Space Life's SS12 figurine collection."
 	icon_state = "chef"
 	toysay = "I swear it's not human meat."
 
-/obj/item/toy/figure/chemist
+/obj/item/toy/figure/crew/chemist
 	name = "Chemist action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1520]:
```suggestion
	name = "\improper Chemist action figure"
```

File: code/game/objects/items/toys.dm [hunk 8]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The legally dubious Chemist, from Space Life's SS12 figurine collection."
 	icon_state = "chemist"
 	toysay = "Get your pills!"
 
-/obj/item/toy/figure/clown
+/obj/item/toy/figure/crew/clown
 	name = "Clown action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1526]:
```suggestion
	name = "\improper Clown action figure"
```

File: code/game/objects/items/toys.dm [hunk 9]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The mischevious Clown, from Space Life's SS12 figurine collection."
 	icon_state = "clown"
 	toysay = "Honk!"
 
-/obj/item/toy/figure/ian
+/obj/item/toy/figure/crew/ian
 	name = "Ian action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1532]:
```suggestion
	name = "\improper Ian action figure"
```

File: code/game/objects/items/toys.dm [hunk 10]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The adorable corgi, from Space Life's SS12 figurine collection."
 	icon_state = "ian"
 	toysay = "Arf!"
 
-/obj/item/toy/figure/detective
+/obj/item/toy/figure/crew/detective
 	name = "Detective action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1538]:
```suggestion
	name = "\improper Detective action figure"
```

File: code/game/objects/items/toys.dm [hunk 11]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The clever detective, from Space Life's SS12 figurine collection."
 	icon_state = "detective"
 	toysay = "This airlock has grey jumpsuit and insulated glove fibers on it."
 
-/obj/item/toy/figure/dsquad
+/obj/item/toy/figure/crew/dsquad
 	name = "Death Squad Officer action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1544]:
```suggestion
	name = "\improper Death Squad Officer action figure"
```

File: code/game/objects/items/toys.dm [hunk 12]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "It's a member of the DeathSquad, a TV drama where loose-cannon ERT officers face up against the threats of the galaxy! It's from Space Life's special edition SS12 figurine collection."
 	icon_state = "dsquad"
 	toysay = "Eliminate all threats!"
 
-/obj/item/toy/figure/engineer
+/obj/item/toy/figure/crew/engineer
 	name = "Engineer action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1550]:
```suggestion
	name = "\improper Engineer action figure"
```

File: code/game/objects/items/toys.dm [hunk 13]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The frantic engineer, from Space Life's SS12 figurine collection."
 	icon_state = "engineer"
 	toysay = "Oh god, the singularity is loose!"
 
-/obj/item/toy/figure/geneticist
+/obj/item/toy/figure/crew/geneticist
 	name = "Geneticist action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1556]:
```suggestion
	name = "\improper Geneticist action figure"
```

File: code/game/objects/items/toys.dm [hunk 14]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The balding geneticist, from Space Life's SS12 figurine collection."
 	icon_state = "geneticist"
 	toysay = "I'm not qualified for this job."
 
-/obj/item/toy/figure/hop
+/obj/item/toy/figure/crew/hop
 	name = "Head of Personnel action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1562]:
```suggestion
	name = "\improper Head of Personnel action figure"
```

File: code/game/objects/items/toys.dm [hunk 15]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The officious Head of Personnel, from Space Life's SS12 figurine collection."
 	icon_state = "hop"
 	toysay = "Papers, please!"
 
-/obj/item/toy/figure/hos
+/obj/item/toy/figure/crew/hos
 	name = "Head of Security action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1568]:
```suggestion
	name = "\improper Head of Security action figure"
```

File: code/game/objects/items/toys.dm [hunk 16]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The bloodlust-filled Head of Security, from Space Life's SS12 figurine collection."
 	icon_state = "hos"
 	toysay = "Space law? What?"
 
-/obj/item/toy/figure/qm
+/obj/item/toy/figure/crew/qm
 	name = "Quartermaster action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1574]:
```suggestion
	name = "\improper Quartermaster action figure"
```

File: code/game/objects/items/toys.dm [hunk 17]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The nationalistic Quartermaster, from Space Life's SS12 figurine collection."
 	icon_state = "qm"
 	toysay = "Hail Cargonia!"
 
-/obj/item/toy/figure/janitor
+/obj/item/toy/figure/crew/janitor
 	name = "Janitor action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1580]:
```suggestion
	name = "\improper Janitor action figure"
```

File: code/game/objects/items/toys.dm [hunk 18]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The water-using Janitor, from Space Life's SS12 figurine collection."
 	icon_state = "janitor"
 	toysay = "Look at the signs, you idiot."
 
-/obj/item/toy/figure/lawyer
+/obj/item/toy/figure/crew/lawyer
 	name = "Internal Affairs Agent action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1586]:
```suggestion
	name = "\improper Internal Affairs Agent action figure"
```

File: code/game/objects/items/toys.dm [hunk 19]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The unappreciated Internal Affairs Agent, from Space Life's SS12 figurine collection."
 	icon_state = "lawyer"
 	toysay = "Standard Operating Procedure says they're guilty! Hacking is proof they're an Enemy of the Corporation!"
 
-/obj/item/toy/figure/librarian
+/obj/item/toy/figure/crew/librarian
 	name = "Librarian action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1592]:
```suggestion
	name = "\improper Librarian action figure"
```

File: code/game/objects/items/toys.dm [hunk 20]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The quiet Librarian, from Space Life's SS12 figurine collection."
 	icon_state = "librarian"
 	toysay = "One day while..."
 
-/obj/item/toy/figure/md
+/obj/item/toy/figure/crew/md
 	name = "Medical Doctor action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1598]:
```suggestion
	name = "\improper Medical Doctor action figure"
```

File: code/game/objects/items/toys.dm [hunk 21]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The stressed-out doctor, from Space Life's SS12 figurine collection."
 	icon_state = "md"
 	toysay = "The patient is already dead!"
 
-/obj/item/toy/figure/mime
+/obj/item/toy/figure/crew/mime
 	name = "Mime action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1604]:
```suggestion
	name = "\improper Mime action figure"
```

File: code/game/objects/items/toys.dm [hunk 22]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "... from Space Life's SS12 figurine collection."
 	icon_state = "mime"
 	toysay = "..."
 
-/obj/item/toy/figure/miner
+/obj/item/toy/figure/crew/miner
 	name = "Shaft Miner action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1610]:
```suggestion
	name = "\improper Shaft Miner action figure"
```

File: code/game/objects/items/toys.dm [hunk 23]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The gun-toting Shaft Miner, from Space Life's SS12 figurine collection."
 	icon_state = "miner"
 	toysay = "Oh god it's eating my intestines!"
 
-/obj/item/toy/figure/ninja
+/obj/item/toy/figure/crew/ninja
 	name = "Ninja action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1616]:
```suggestion
	name = "\improper Ninja action figure"
```

File: code/game/objects/items/toys.dm [hunk 24]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "It's the mysterious ninja! It's from Space Life's special edition SS12 figurine collection."
 	icon_state = "ninja"
 	toysay = "Oh god! Stop shooting, I'm friendly!"
 
-/obj/item/toy/figure/wizard
+/obj/item/toy/figure/crew/wizard
 	name = "Wizard action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1622]:
```suggestion
	name = "\improper Wizard action figure"
```

File: code/game/objects/items/toys.dm [hunk 25]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "It's the deadly, spell-slinging wizard! It's from Space Life's special edition SS12 figurine collection."
 	icon_state = "wizard"
 	toysay = "Ei Nath!"
 
-/obj/item/toy/figure/rd
+/obj/item/toy/figure/crew/rd
 	name = "Research Director action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1628]:
```suggestion
	name = "\improper Research Director action figure"
```

File: code/game/objects/items/toys.dm [hunk 26]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The ambitious RD, from Space Life's SS12 figurine collection."
 	icon_state = "rd"
 	toysay = "Blowing all of the borgs!"
 
-/obj/item/toy/figure/roboticist
+/obj/item/toy/figure/crew/roboticist
 	name = "Roboticist action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1634]:
```suggestion
	name = "\improper Roboticist action figure"
```

File: code/game/objects/items/toys.dm [hunk 27]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The skillful Roboticist, from Space Life's SS12 figurine collection."
 	icon_state = "roboticist"
 	toysay = "He asked to be borged!"
 
-/obj/item/toy/figure/scientist
+/obj/item/toy/figure/crew/scientist
 	name = "Scientist action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1640]:
```suggestion
	name = "\improper Scientist action figure"
```

File: code/game/objects/items/toys.dm [hunk 28]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The mad Scientist, from Space Life's SS12 figurine collection."
 	icon_state = "scientist"
 	toysay = "Someone else must have made those bombs!"
 
-/obj/item/toy/figure/syndie
+/obj/item/toy/figure/crew/syndie
 	name = "Nuclear Operative action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1646]:
```suggestion
	name = "\improper Nuclear Operative action figure"
```

File: code/game/objects/items/toys.dm [hunk 29]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "It's the red-suited Nuclear Operative! It's from Space Life's special edition SS12 figurine collection."
 	icon_state = "syndie"
 	toysay = "Get that fucking disk!"
 
-/obj/item/toy/figure/secofficer
+/obj/item/toy/figure/crew/secofficer
 	name = "Security Officer action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1652]:
```suggestion
	name = "\improper Security Officer action figure"
```

File: code/game/objects/items/toys.dm [hunk 30]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The power-tripping Security Officer, from Space Life's SS12 figurine collection."
 	icon_state = "secofficer"
 	toysay = "I am the law!"
 
-/obj/item/toy/figure/virologist
+/obj/item/toy/figure/crew/virologist
 	name = "Virologist action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1658]:
```suggestion
	name = "\improper Virologist action figure"
```

File: code/game/objects/items/toys.dm [hunk 31]
```diff
@@ -1570,243 +1413,387 @@
...
 	desc = "The pandemic-starting Virologist, from Space Life's SS12 figurine collection."
 	icon_state = "virologist"
 	toysay = "It's not my virus!"
 
-/obj/item/toy/figure/warden
+/obj/item/toy/figure/crew/warden
 	name = "Warden action figure"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1664]:
```suggestion
	name = "\improper Warden action figure"
```

File: code/game/objects/items/toys.dm [hunk 32]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "MEGA presents the new Xenos Isolated action figure! Comes complete with realistic sounds! Pull back string to use."
+	icon_state = "toy_xeno"
+	bubble_icon = "alien"
+	cooldown_time = 5 SECONDS
+
+
+/obj/item/toy/figure/xeno/activate(mob/user)
+	user.visible_message("<span class='notice'>[user] pulls back the string on [src].</span>")
+	icon_state = "[initial(icon_state)]_used"
+	addtimer(CALLBACK(src, .proc/hiss), 5)
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1684]:
```suggestion
	addtimer(CALLBACK(src, .proc/hiss), 0.5 SECONDS)
```

File: code/game/objects/items/toys.dm [hunk 33]
```diff
@@ -1570,243 +1413,387 @@
...
+
+/obj/item/toy/figure/xeno/proc/hiss()
+	atom_say("Hiss!")
+	var/list/possible_sounds = list('sound/voice/hiss1.ogg', 'sound/voice/hiss2.ogg', 'sound/voice/hiss3.ogg', 'sound/voice/hiss4.ogg')
+	playsound(get_turf(src), pick(possible_sounds), 50, 1)
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1689]:
```suggestion
	playsound(src, get_sfx("hiss"), 50, TRUE)
```

File: code/game/objects/items/toys.dm [hunk 34]
```diff
@@ -1570,243 +1413,387 @@
...
+	addtimer(CALLBACK(src, .proc/reset_icon), 45)
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1690]:
```suggestion
	addtimer(CALLBACK(src, .proc/reset_icon), 4.5 SECONDS)
```

File: code/game/objects/items/toys.dm [hunk 35]
```diff
@@ -1570,243 +1413,387 @@
...
+
+/obj/item/toy/figure/owl
+	name = "owl action figure"
+	desc = "An action figure modeled after 'The Owl', defender of justice."
+	icon_state = "owlprize"
+
+/obj/item/toy/figure/owl/activate(mob/user)
+	var/message = pick("You won't get away this time, Griffin!", "Stop right there, criminal!", "Hoot! Hoot!", "I am the night!")
+	to_chat(user, "<span class='notice'>You pull the string on the [src].</span>")
+	playsound(user, 'sound/creatures/hoot.ogg', 25, 1)
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1706]:
```suggestion
	playsound(src, 'sound/creatures/hoot.ogg', 25, TRUE)
```

File: code/game/objects/items/toys.dm [hunk 36]
```diff
@@ -1570,243 +1413,387 @@
...
+	to_chat(user, "<span class='warning'>The string on [src] hasn't rewound all the way!</span>")
+
+/obj/item/toy/figure/owl
+	name = "owl action figure"
+	desc = "An action figure modeled after 'The Owl', defender of justice."
+	icon_state = "owlprize"
+
+/obj/item/toy/figure/owl/activate(mob/user)
+	var/message = pick("You won't get away this time, Griffin!", "Stop right there, criminal!", "Hoot! Hoot!", "I am the night!")
+	to_chat(user, "<span class='notice'>You pull the string on the [src].</span>")
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1705]:
```suggestion
	to_chat(user, "<span class='notice'>You pull the string on [src].</span>")
```

File: code/game/objects/items/toys.dm [hunk 37]
```diff
@@ -1570,243 +1413,387 @@
...
+
+/obj/item/toy/figure/griffin
+	name = "griffin action figure"
+	desc = "An action figure modeled after 'The Griffin', criminal mastermind."
+	icon_state = "griffinprize"
+
+
+/obj/item/toy/figure/griffin/activate(mob/user)
+	var/message = pick("You can't stop me, Owl!", "My plan is flawless! The vault is mine!", "Caaaawwww!", "You will never catch me!")
+	to_chat(user, "<span class='notice'>You pull the string on the [src].</span>")
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1717]:
```suggestion
	to_chat(user, "<span class='notice'>You pull the string on [src].</span>")
```

File: code/game/objects/items/toys.dm [hunk 38]
```diff
@@ -1570,243 +1413,387 @@
...
+	playsound(user, 'sound/creatures/caw.ogg', 25, 1)
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1718]:
```suggestion
	playsound(src, 'sound/creatures/caw.ogg', 25, TRUE)
```

File: code/game/objects/items/toys.dm [hunk 39]
```diff
@@ -1570,243 +1413,387 @@
...
+ * Mech prizes
+ */
+/obj/item/toy/figure/mech
+	icon_state = "ripleytoy"
+	cooldown_time = 8
+
+//all credit to skasi for toy mech fun ideas
+/obj/item/toy/figure/mech/activate(mob/user)
+	to_chat(user, "<span class='notice'>You play with [src].</span>")
+	playsound(user, 'sound/mecha/mechstep.ogg', 20, 1)
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1731]:
```suggestion
	playsound(src, 'sound/mecha/mechstep.ogg', 20, TRUE)
```

File: code/game/objects/items/toys.dm [hunk 40]
```diff
@@ -1570,243 +1413,387 @@
...
+	name = "Random Mech Prize"
+	desc = "This is a random prize"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "ripleytoy"
+
+/obj/random/mech/item_to_spawn()
+	return pick(subtypesof(/obj/item/toy/figure/mech)) //exclude the base type.
+
+/obj/item/toy/figure/mech/ripley
+	name = "toy ripley"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1743]:
```suggestion
	name = "toy Ripley"
```

File: code/game/objects/items/toys.dm [hunk 41]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 1/11. This one is a ripley, a mining and engineering mecha."
+
+/obj/item/toy/figure/mech/fireripley
+	name = "toy firefighting ripley"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1747]:
```suggestion
	name = "toy Firefighting Ripley"
```

File: code/game/objects/items/toys.dm [hunk 42]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 2/11. This one is a firefighter ripley, a fireproof mining and engineering mecha."
+	icon_state = "fireripleytoy"
+
+/obj/item/toy/figure/mech/deathripley
+	name = "toy deathsquad ripley"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1752]:
```suggestion
	name = "toy Death Squad Ripley"
```

File: code/game/objects/items/toys.dm [hunk 43]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 3/11. This one is the black ripley used by the hero of DeathSquad, that TV drama about loose-cannon ERT officers!"
+	icon_state = "deathripleytoy"
+
+/obj/item/toy/figure/mech/gygax
+	name = "toy gygax"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1757]:
```suggestion
	name = "toy Gygax"
```

File: code/game/objects/items/toys.dm [hunk 44]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 4/11. This one is the speedy gygax combat mecha. Zoom zoom, pew pew!"
+	icon_state = "gygaxtoy"
+
+/obj/item/toy/figure/mech/durand
+	name = "toy durand"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1762]:
```suggestion
	name = "toy Durand"
```

File: code/game/objects/items/toys.dm [hunk 45]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 5/11. This one is the heavy durand combat mecha. Stomp stomp!"
+	icon_state = "durandprize"
+
+/obj/item/toy/figure/mech/honk
+	name = "toy H.O.N.K."
+	desc = "Mini-Mecha action figure! Collect them all! 6/11. This one is the infamous H.O.N.K mech!"
+	icon_state = "honkprize"
+
+/obj/item/toy/figure/mech/marauder
+	name = "toy marauder"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1772]:
```suggestion
	name = "toy Marauder"
```

File: code/game/objects/items/toys.dm [hunk 46]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 7/11. This one is the powerful marauder combat mecha! Run for cover!"
+	icon_state = "marauderprize"
+
+/obj/item/toy/figure/mech/seraph
+	name = "toy seraph"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1777]:
```suggestion
	name = "toy Seraph"
```

File: code/game/objects/items/toys.dm [hunk 47]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 8/11. This one is the powerful seraph combat mecha! Someone's in trouble!"
+	icon_state = "seraphprize"
+
+/obj/item/toy/figure/mech/mauler
+	name = "toy mauler"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1782]:
```suggestion
	name = "toy Mauler"
```

File: code/game/objects/items/toys.dm [hunk 48]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 9/11. This one is the deadly mauler combat mecha! Look out!"
+	icon_state = "maulerprize"
+
+/obj/item/toy/figure/mech/odysseus
+	name = "toy odysseus"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1787]:
```suggestion
	name = "toy Odysseus"
```

File: code/game/objects/items/toys.dm [hunk 49]
```diff
@@ -1570,243 +1413,387 @@
...
+	desc = "Mini-Mecha action figure! Collect them all! 10/11. This one is the spindly, syringe-firing odysseus medical mecha."
+	icon_state = "odysseusprize"
+
+/obj/item/toy/figure/mech/phazon
+	name = "toy phazon"
```
Comment (dearmochi) [inline code/game/objects/items/toys.dm:1792]:
```suggestion
	name = "toy Phazon"
```

**Review by dearmochi — APPROVED** [review]

File: code/game/objects/items/toys.dm [hunk 50]
```diff
@@ -1570,243 +1413,386 @@
...
+	. = ..()
 	desc = "A \"Space Life\" brand [name]"
 
-/obj/item/toy/figure/attack_self(mob/user as mob)
-	if(cooldown < world.time)
-		cooldown = (world.time + 30) //3 second cooldown
-		user.visible_message("<span class='notice'>[bicon(src)] The [src] says \"[toysay]\".</span>")
-		playsound(user, 'sound/machines/click.ogg', 20, 1)
+/obj/item/toy/figure/crew/activate(mob/user)
+	user.visible_message("<span class='notice'>[bicon(src)] [src] says \"[toysay]\".</span>")
```
Comment (Fox-McCloud) [inline code/game/objects/items/toys.dm:1450]: These should `atom_say`, IMO.

File: code/game/objects/items/toys.dm [hunk 51]
```diff
@@ -1570,243 +1413,386 @@
...
+/obj/item/toy/figure/owl
+	name = "owl action figure"
+	desc = "An action figure modeled after 'The Owl', defender of justice."
+	icon_state = "owlprize"
+
+/obj/item/toy/figure/owl/activate(mob/user)
+	var/message = pick("You won't get away this time, Griffin!", "Stop right there, criminal!", "Hoot! Hoot!", "I am the night!")
+	to_chat(user, "<span class='notice'>You pull the string on [src].</span>")
+	playsound(src, 'sound/creatures/hoot.ogg', 25, TRUE)
+	visible_message("<span class='danger'>[bicon(src)] [message]</span>")
```
Comment (Fox-McCloud) [inline code/game/objects/items/toys.dm:1706]: here and other similar things elsewhere

File: code/game/objects/items/toys.dm (hunk 50 above)
Comment (farie82) [reply code/game/objects/items/toys.dm:1450]: Will change

File: code/game/objects/items/toys.dm [hunk 52]
```diff
@@ -1570,243 +1413,386 @@
...
+
+/obj/item/toy/figure/proc/on_cooldown(mob/user)
+	return
+
+/obj/item/toy/figure/crew
+	var/toysay = "What the fuck did you do?"
+
+/obj/item/toy/figure/crew/Initialize(mapload)
+	. = ..()
 	desc = "A \"Space Life\" brand [name]"
```
Comment (ExusA) [inline code/game/objects/items/toys.dm:1447]: Could you remove this line? All the figurines actually do have descriptions, but they get replaced by this.

Comment (farie82) [reply code/game/objects/items/toys.dm:1447]: Oh, interesting didn't even notice that when making the PR. Will change!

File: code/modules/arcade/arcade_prize.dm [hunk 53]
```diff
@@ -2,19 +2,19 @@
...
+/obj/item/toy/figure/mechball
 	name = "prize ball"
 	desc = "A toy is a toy, but a prize ball could be anything! It could even be a toy!"
 	icon = 'icons/obj/arcade.dmi'
 	icon_state = "prizeball_1"
 	var/opening = 0
 	var/possible_contents = list(/obj/random/carp_plushie, /obj/random/plushie, /obj/random/figure, /obj/item/toy/eight_ball, /obj/item/stack/tickets)
 
-/obj/item/toy/prizeball/New()
+/obj/item/toy/figure/mechball/New()
```
Comment (AffectedArc07) [inline code/modules/arcade/arcade_prize.dm:13]: Same here

File: code/modules/arcade/arcade_prize.dm [hunk 54]
```diff
@@ -2,19 +2,19 @@
 * Prize balls
 * Prize tickets
 */
-/obj/item/toy/prizeball
+/obj/item/toy/figure/mechball
```
Comment (AffectedArc07) [inline code/modules/arcade/arcade_prize.dm:5]: Why did you make this type `mechball` when it has other types that arent mechs

image

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/modules/arcade/arcade_prize.dm (hunk 54 above)
Comment (farie82) [reply code/modules/arcade/arcade_prize.dm:5]: Good point. Seems I overlooked that one. Will change it to something more sensible

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**